### PR TITLE
style(db): rename constraint names to follow sql style guide

### PR DESCRIPTION
**Pull request by A-440Hz**

### CODE REVIEW

**Comment by tmessi** [comment]
Would you mind bringing this up to date with main? I am curious to see the new schema-diff run against this.

File: internal/db/schema/migrations/oss/postgres/66/01_rename_constraints.up.sql [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            and pa.attrelid = const.table_num
+        ), '_'), '');
+    end loop;
+    if length(const.attrs_concat) > 0 and length(const.new_name) <= 63
+      then
+        execute format('alter table %I rename constraint %I to %I', const.table_name, const.constraint_name, const.new_name);
+    end if;
+  end loop;
+  end;
+  $$ language plpgsql;
```
Comment (tmessi) [inline internal/db/schema/migrations/oss/postgres/66/01_rename_constraints.up.sql:51]: I think some formatting would help with the readability of this, ie:

```suggestion
  do $$
  declare
    const record;
    i     int;
  begin
    for const in
        select pc.conname  as constraint_name,
               pgc.relname as table_name,
               pgc.relname || '_' || (
                select string_agg(attname, '_')
                  from (
                    select pa.attname
                      from pg_catalog.pg_attribute pa
                     where pa.attrelid = pc.conrelid
                       and pa.attnum   = any(pc.conkey)
                  order by pa.attname asc
                 ) as attname
               ) || '_uq'  as new_name,
               pc.conrelid as table_num,
               pc.conkey   as attr_keys,
               regexp_replace(regexp_replace(pc.conname, '^' || pgc.relname || '_', ''), 'uq$', '') as attrs_concat
          from pg_catalog.pg_constraint pc
          join pg_catalog.pg_class      pgc on pc.conrelid = pgc.oid
         where pgc.relname not like 'pg_%'
           and contype = 'u'
      order by pgc.relname, pc.conname

      loop foreach i in array const.attr_keys
        loop const.attrs_concat := regexp_replace(
          const.attrs_concat,
          concat((
              select pa.attname
                from pg_catalog.pg_attribute pa
               where pa.attnum   = i
                 and pa.attrelid = const.table_num
            ), '_'), '');
        end loop;
        if length(const.attrs_concat) > 0 and length(const.new_name) <= 63
          then
            execute format('alter table %I rename constraint %I to %I', const.table_name, const.constraint_name, const.new_name);
        end if;
      end loop;
  end;
  $$ language plpgsql;
```

File: internal/db/schema/migrations/oss/postgres/66/01_rename_constraints.up.sql [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    if length(const.new_name) <= 63
+      then
+      begin
+        execute format('alter table if exists %I rename constraint %I to %I', const.con_table, const.constraint_name, const.new_name);
+      exception when others then
+      end;
+    end if;
+  end loop;
+  end;
+  $$ language plpgsql;
```
Comment (tmessi) [inline internal/db/schema/migrations/oss/postgres/66/01_rename_constraints.up.sql:99]: Similarly here:

```suggestion
  do $$
  declare
    const record;
    i     int;
  begin
  for const in
      select pc.conname         as constraint_name,
             pgc.relname        as table_name,
             pgc.relname || '_' as new_name,
             pc.conrelid        as conrelid,
             pc.conkey          as conkey,
             pgc2.relname       as con_table
        from pg_catalog.pg_constraint pc
        join pg_catalog.pg_class      pgc on  pc.confrelid = pgc.oid
        join pg_catalog.pg_class      pgc2 on pc.conrelid  = pgc2.oid
       where pgc.relname not like 'pg_%'
         and contype = 'f'
         and (pc.conname !~ ('^' || pgc.relname || '_')
          or pc.conname !~ 'fkey[1]?$')
    order by pgc.relname, pc.conname
  loop
    i = (
      select max(x)
        from unnest(const.conkey) as x
    );
    const.new_name = const.new_name || (
      select pa.attname
        from pg_catalog.pg_attribute pa
       where const.conrelid = pa.attrelid
         and pa.attnum      = i
    ) || '_fkey';
    if length(const.new_name) <= 63
      then
      begin
        execute format('alter table if exists %I rename constraint %I to %I', const.con_table, const.constraint_name, const.new_name);
      exception when others then
      end;
    end if;
  end loop;
  end;
  $$ language plpgsql;
```

**Review by tmessi — CHANGES_REQUESTED** [review]
The sql diff looks good, and yields the same result for postges 11, 12, 13, 14, and 15. I just have some requested changes to the formatting of the new migration.

**Review by mgaffney — COMMENTED** [review]
The naming pattern postgres uses for naming unnamed constraints can change (and has changed) between major releases. Which versions of postgres have we tested this against?

**Comment by A-440Hz** [comment]
@mgaffney the diffs have been tested against postgres 11, 12, 13, 14, and 15

**Comment by mgaffney** [comment]
> @mgaffney the diffs have been tested against postgres 11, 12, 13, 14, and 15

Sorry, I don't know what that means. Is there a link that explains the who, what, when, how of the diffs?

**Comment by A-440Hz** [comment]
> > @mgaffney the diffs have been tested against postgres 11, 12, 13, 14, and 15
> 
> Sorry, I don't know what that means. Is there a link that explains the who, what, when, how of the diffs?

My bad. My statement was unclear.
Tim has told me he changed the `PG_DOCKER_TAG` values in the `internal/db/sqltest/Makefile` for the respective postgres versions, ran the `schema-diff.sh` script to generate a diff file like in the GHA output above, and then diffed the diffs against each other.
Currently I am having trouble getting the script to run locally in my setup, but I will work on successfully reproducing his steps locally and producing 5 diffs of 11v12, 11v13, 11v14, 11v15.

**Comment by A-440Hz** [comment]
I ran the script locally with `PG_DOCKER_TAG` of [11-15]-alpine and [11, 15]-bullseye. Each time, it generated the same result as the github-actions bot above

**Review by mgaffney — APPROVED** [review]

**Pull request merged**